### DISCUSSION TITLE
fetchQuery() resets cache

### DISCUSSION THREAD

**Discussion by t0x1c123** [TanStack/query#2024] [Q&A]
Hi, after long investigation I came to result that when I use fetchQuery all my cached date (active and also inactive) are removed from cache. So for example:

I have 2 pages. When I go from page 1 to page 2 I call fetchQuery. On page 2 I call three useQuery(). I go back to page 1 all three queries are in state inactive. Then I go again to page 2 I call fetchQuery and my cache is cleared and all three useQuery are fetched again although they dont need to be because I had them in cache.

I'm not sure if it's bug or feature 😄 

I've got version 3.7.1

**Comment by TkDodo** [comment]
could you create a codesandbox example please? It sounds more like you have `new QueryClient()` _inside_ your page1, which will create a new cache when that component renders. But just guessing here, have to see some code please.

**Reply by t0x1c123** [reply]
It's little bit complicated than my example :D But I use

``const queryClient = useQueryClient();``

It's inside HOC component that wraps parent of Page 1 and Page2. And between transition of page 1 and 2 is called function inside that HOC component. But I'm not calling new ``QueryClient()`` only `` useQueryClient();``. If it's needed I'll try to create some lightweight example in codesandbox.

**Reply by TkDodo** [reply]
`userQueryClient` is totally fine, that just gives you the queryClient via react context.

yes, please create an example. `fetchQuery` is also called in `useQuery`, so I really doubt that this is the one that clears your cache :)
